pkg/exporter/probe/procfd/procfd.go

```diff
@@ -0,0 +1,116 @@
+package procfd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/alibaba/kubeskoop/pkg/exporter/nettop"
+
+	"golang.org/x/exp/slog"
+)
+
+const (
+	ModuleName = "Procfd" // nolint
+)
+
+var (
+	probe = &ProcFD{}
+
+	FDMetrics = []string{"OpenFd", "OpenSocket"}
+)
+
+type ProcFD struct {
+}
+
+func GetProbe() *ProcFD {
+	return probe
+}
+
+func (s *ProcFD) Close() error {
+	return nil
+}
+
+func (s *ProcFD) Start(_ context.Context) {
+}
+
+func (s *ProcFD) Ready() bool {
+	return true
+}
+
+func (s *ProcFD) Name() string {
+	return ModuleName
+}
+
+func (s *ProcFD) GetMetricNames() []string {
+	res := []string{}
+	for _, m := range FDMetrics {
+		res = append(res, metricUniqueID("fd", m))
+	}
+	return res
+}
+
+func (s *ProcFD) Collect(ctx context.Context) (map[string]map[uint32]uint64, error) {
+	ets := nettop.GetAllEntity()
+	if len(ets) == 0 {
+		slog.Ctx(ctx).Info("collect", "mod", ModuleName, "ignore", "no entity found")
+	}
+	return getAllProcessFd(ets)
+}
+
+func metricUniqueID(subject string, m string) string {
+	return fmt.Sprintf("%s%s", subject, strings.ToLower(m))
+}
+
+func getAllProcessFd(nslist []*nettop.Entity) (map[string]map[uint32]uint64, error) {
+	resMap := make(map[string]map[uint32]uint64)
+	for _, metricname := range FDMetrics {
+		resMap[metricUniqueID("fd", metricname)] = map[uint32]uint64{}
+	}
+	for _, nslogic := range nslist {
+		nsprocfd := map[string]struct{}{}
+		nsprocfsock := map[string]struct{}{}
+		for _, idx := range nslogic.GetPids() {
+			procfds, err := getProcessFdStat(idx)
+			if err != nil {
```

Comment:
If `pid` not exist, continue to collect others pid, instead of return error directly.

Reply:
fine, fixed now.